src/benchmark/operators/std_sort_benchmark.cpp

```diff
@@ -0,0 +1,110 @@
+#include <algorithm>
+#include <random>
+
+#include "benchmark/benchmark.h"
+#include "synthetic_table_generator.hpp"
+#include "types.hpp"
+
+#define SIZE 40'000
+
+namespace opossum {
+
+static void BM_StdBaseInt(benchmark::State& state) {
+  std::random_device rd;
+  std::mt19937 mt(rd());
+  std::uniform_int_distribution<int> dist;
+
+  for (auto _ : state) {
+    pmr_vector<int> v;
+    v.reserve(state.range(0));
+    for (int i = 0; i < state.range(0); i++) {
+      v.push_back(dist(mt));
+    }
+  }
+}
+BENCHMARK(BM_StdBaseInt)->Arg(SIZE);
+
+static void BM_StdSortInt(benchmark::State& state) {
+  std::random_device rd;
+  std::mt19937 mt(rd());
+  std::uniform_int_distribution<int> dist;
+
+  for (auto _ : state) {
+    pmr_vector<int> v;
+    v.reserve(state.range(0));
+    for (int i = 0; i < state.range(0); i++) {
+      v.push_back(dist(mt));
+    }
+
+    std::sort(v.begin(), v.end());
```

Comment:
Please consider benchmark actual data as it would occur in Hyrise. That means both actual value vectors with different encodings (I guess you have that on your roadmap already), but most importantly (and that's why I am adding the review here) you want to cover sorting reference tables. Except for the data loading part in Hyrise for TPC-H (which can be considered an edge case), almost all sorting ever taking place in Hyrise will be on tables which have already been filtered.
For such reference tables, it will be VERY interesting to see the effects of a full pos list (e.g., after a validate that does not throw out rows), a sorted but not sequential pos list (e.g., a filter that removed 50% of tuples), and a "random" pos list (e.g, generated via a join, probably the most important use case in Hyrise).

Reply:
Hi @Bouncner and thanks for your feedback! Actually this PR (which is also not a "real" PR but only a Draft PR) was only meant as a way to put our results regarding the out-of-context low-level benchmarks for comparing `std::sort` to `std::stable_sort` somewhere visible. The benchmarks are currently **not** intended to be included in our fork's master branch, even less in the hyrise upstream – thus, we focused less on using actual Hyrise data structures and formats (as opposed to in the actual sort_benchmark.cpp micro benchmarks, where we put attention to those Hyrise-specific details) here in this Draft PR and instead only wanted to compare the general performance of `std::sort` to `std::stable_sort`. :Bligh

Reply:
Yeah sure, and this benchmarks make perfect sense. Nonetheless, please consider the issues brought up for the coming weeks.